select/index.js

```diff
@@ -0,0 +1,313 @@
+            case 'Enter':
+            case ' ':
+                this.selectOption(currentIndex);
+                this.closeList();
+                return;
+            case 'Escape':
+                this.closeList();
+                return;
+            default:
+                return;
+        }
+    
+        event.preventDefault();
+        this.focusItem(newIndex);
+    };
+    
+
+    toggleList() {
+        this.isOpen = !this.isOpen;
+    
+        const list = this.querySelector('.irnmn-select__list');
+        const select = this.querySelector('.irnmn-select');
+        const header = this.querySelector('.irnmn-select__header');
+    
+        if (this.isOpen) {
+            this.determineDropdownPosition();
+            this.focusItem(this.selectedOption !== null ? this.selectedOption : 0);
+        } else {
+            list.classList.remove('open-upwards');
+        }
+    
+        list.classList.toggle('irnmn-select__list--open', this.isOpen);
+        select.setAttribute('aria-expanded', this.isOpen);
+    
+        if (this.isOpen) {
+            this.focusItem(this.selectedOption !== null ? this.selectedOption : 0);
+        } else {
+            header.focus();
+        }
+    }    
+
+    closeList() {
+        this.isOpen = false;
+        const list = this.querySelector('.irnmn-select__list');
+        list.classList.remove('irnmn-select__list--open');
+        this.querySelector('.irnmn-select').setAttribute(
+            'aria-expanded',
+            'false',
+        );
+    }
+
+    determineDropdownPosition() {
+        const rect = this.getBoundingClientRect();
+        const viewportHeight = window.innerHeight;
+        const spaceBelow = viewportHeight - rect.bottom;
+        const spaceAbove = rect.top;
+    
+        const list = this.querySelector('.irnmn-select__list');
+    
+        // Temporarily show the dropdown to calculate its height - Need to find a cleaner way!!
+        list.style.display = 'block';  // Make it visible to measure height
+        const dropdownHeight = list.offsetHeight;
+        list.style.display = '';  // Reset the display to its original value
+    
+        // Only add 'open-upwards' if there is not enough space below but there is enough space above
+        if (spaceBelow < dropdownHeight && spaceAbove > dropdownHeight) {
+            list.classList.add('open-upwards');
+        } else {
+            list.classList.remove('open-upwards');
+        }
+    }    
+    
+
+    selectOption(index) {
+        this.selectedOption = index;
+        this.render();
+        const selectedOption = this.options[index];
+        this.dispatchEvent(
+            new CustomEvent('optionSelected', {
+                detail: {
+                    option: selectedOption.name,
+                    value: selectedOption.value,
+                    index: index,
+                },
+                bubbles: true,
+            }),
+        );
+    }
+
+    focusItem(index) {
```

Comment:
This could live in that extenral file for accesbility

Reply:
@samuelsilvapt please see answer above regarding handleKeydown